GCEWindowsAgent/metadata.go

```diff
@@ -36,30 +36,35 @@ var (
 	etag              = defaultEtag
 )
 
-type metadataJSON struct {
-	Instance instanceJSON
-	Project  projectJSON
+type metadata struct {
+	Instance
+	Project
 }
 
-type instanceJSON struct {
-	Attributes        attributesJSON
-	NetworkInterfaces []networkInterfacesJSON
+type VirtualClock struct {
+	DriftToken int
 }
 
-type networkInterfacesJSON struct {
+type Instance struct {
+	Attributes
+	NetworkInterfaces []networkInterfaces
+	VirtualClock
+}
+
+type networkInterfaces struct {
 	ForwardedIps      []string
 	TargetInstanceIps []string
 	IPAliases         []string
 	Mac               string
 }
 
-type projectJSON struct {
-	Attributes attributesJSON
-	ProjectID  string `json:"projectId"`
+type Project struct {
```

Comment:
None of these types need to be public

Reply:
easier to read the struct by simple embedding. json needs public attrs to do unmarshaling, and so we get a lot of `Private privateJSON` repeats for no reason. think this is a case where it makes sense to have public types.